clang/tools/dpct/codepin/codepin-report.py

```diff
@@ -20,11 +21,16 @@
 TIME_ELAPSED = "Elapse Time(ms)"
 ERROR_MATCH_PATTERN = "Unable to find the corresponding serialization function"
 CODEPIN_REPORT_FILE = os.path.join(os.getcwd(), "CodePin_Report.csv")
-match_checkpoint_num = 0
-dismatch_checkpoint_num = 0
-checkpoint_size = 0
-
 ERROR_CSV_PATTERN = "CUDA Meta Data ID, SYCL Meta Data ID, Type, Detail\n"
+TOLERANCE_FILE = ""
+
+# Reference: https://en.wikipedia.org/wiki/Machine_epsilon
+default_tolerances = {
+    "bf16": 9.77e-04,
+    "fp16": 9.77e-04,
+    "float": 1.19e-07,
```

Comment:
for the end user, a little complex to set the float number here, how about change to set num of ULP instead.

Reply:
I prefer to keep this epsilon value here. As our floating-point comparison is developed by python script. And python code comparing the double type as the default type. Using the ULP to compare the different floating type need more work on it.